src/gromacs/nblib/tests/molecules.cpp

```diff
@@ -198,6 +198,32 @@ TEST(NBlibTest, CanConstructExclusionListFromNamesAndIndicesMixed)
         EXPECT_EQ(exclusions[i], reference[i]);
 }
 
-} // namespace
-} // namespace test
-} // namespace nblib
+TEST(NBlibTest, CanAddHarmonicBond)
+{
+    //! Manually Create Molecule (Water)
+
+    //! 1. Define Atom Type
+    OwAtom   owAtom;
+    AtomType Ow(owAtom.name, owAtom.mass, owAtom.c6, owAtom.c12);
+    HwAtom   hwAtom;
+    AtomType Hw(hwAtom.name, hwAtom.mass, hwAtom.c6, hwAtom.c12);
+
+    //! 2. Define Molecule
+    Molecule water("water");
+
+    water.addAtom(AtomName("Oxygen"), Charge(-0.6), Ow);
+    water.addAtom(AtomName("H1"), Charge(+0.3), Hw);
+    water.addAtom(AtomName("H2"), Charge(+0.3), Hw);
+
+    HarmonicBondType OHbond(EquiDist(0.4), ForceConstant(1.2), BondName("OHbond"));
```

Comment:
I am unsure if we need the name at all. Under what circumstances would it be used?

Reply:
Point. We can consider omitting it. It is for generating a file of the `SimState` object which should be fully human readable.

Reply:
This is a point that could warrant further discussion. It might be nice to have a fancy printer that prints out a whole topology in a nice layout. We could also use this as another way to test that the topology class is doing what we expect.

Reply:
That is certainly useful.

Whether we want to turn it into a JSON-esque file for saving as input files or want to print out a user-readable file that tells the story for diagnostics, having strings to caption objects could be of use.

Reply:
The problem I see is that then we have to have a way to generate strings if none are provided, and possibly also a way to map between names and programmatic access methods. I think we might be able to get away with just using the index in the std::vector for each interaction in the molecule and prefix that with the interaction type. Then we don't need any extra logic for printing than for serialization.

Reply:
Yep. The logic would be similar for serializing and printing.

We could use the type name itself instead of a string. Do we follow the same policy for `AtomType` as well? We could defer the idea of whether we should be getting rid of the strings once we come to a point of writing our serializer/printer.

At the API level, it's merely changing the constructor. So shouldn't be too complicated to edit at a later date.

Reply:
> At the API level, it's merely changing the constructor. So shouldn't be too complicated to edit at a later date.

Agreed